app/controllers/VerifiedEmailController.scala

```diff
@@ -53,6 +55,23 @@ class VerifiedEmailController @Inject() (
     }
   }
 
+  def getVerifiedEmailV2(): Action[AnyContent] = authorisedRequest async {
+    implicit request: RequestWithEori[AnyContent] =>
+      emailRepo.get(request.eori.value).flatMap {
+        case Some(value) => Future.successful(Ok(Json.toJson(value)))
+        case None        => retrieveAndStoreEmail(request.eori.value)
+      }
+  }
+
+  private def retrieveAndStoreEmail(eori: String): Future[Result] =
+    (for {
+      notificationEmail <- OptionT(subscriptionInfoConnector.getSubscriberInformation(eori))
+      result            <- OptionT.liftF(emailRepo.set(eori, notificationEmail).map {
```

Comment:
Below code also could be moved to a private method (i.e. storeEmail) and can be reused (if you are happy to make the change) in getVerifiedEmail as well 

emailRepo.set(eori, notificationEmail).map {
                             case SuccessfulEmail => Ok(Json.toJson(notificationEmail))
                             case _               => InternalServerError
                           }